web/server.js

```diff
@@ -291,6 +292,75 @@ module.exports = function(options) {
         redirectObj.query.logout = true;
         reply.redirect(url.format(redirectObj));
       }
+    },
+    {
+      method: 'GET',
+      path: '/user',
+      config: {
+        auth: false,
+        pre: [
+          {
+            assign: 'requestToken',
+            method: function(request, reply) {
+              var tokenHeader = request.headers.authorization || '';
+              tokenHeader = tokenHeader.split(' ');
+
+              if ( tokenHeader[0] !== 'token' || !tokenHeader[1] ) {
+                return reply(Boom.unauthorized('Missing or invalid authorization header'));
+              }
+
+              reply(tokenHeader[1]);
+            }
+          },
+          {
+            assign: 'token',
+            method: function(request, reply) {
+              oauthDb.lookupAccessToken(request.pre.requestToken, function(err, token) {
+                if ( err ) {
+                  return reply(err);
+                }
+
+                if ( token.expires_at <= Date.now() ) {
+                  return reply(Boom.unauthorized('Expired token'));
+                }
+
+                var tokenScopes = token.scopes.split(' ');
+
+                if ( tokenScopes.indexOf('user') === -1 && tokenScopes.indexOf('email') === -1 ) {
+                  reply(Boom.unauthorized('The token does not have the required scopes'));
+                }
+
+                reply(token);
+              });
+            }
+          },
+          {
+            assign: 'user',
+            method: function(request, reply) {
+              account.getUser(request.pre.token.user_id, function(err, json) {
```

Comment:
Um... adding further logging into the app, it seems like `token.user_id` is set to `undefined` fairly early on, at least as early as `OAuthDB.prototype.generateAuthCode`, which gets passed `undefined` for `userId`...

That function is being called by the handler for `GET /login/oauth/authorize`, which is passing it `request.pre.user.username`, but `request.pre.user` has no such key... In fact, it just looks like this when  logged to the console: `{ set: [Function], clear: [Function], ttl: [Function] }`. So that is weird.